app/src/main/java/com/tft_mvvm/app/ui/activity/MainActivity.kt

```diff
@@ -1,74 +1,31 @@
-import com.tft_mvvm.app.features.champ.model.Champ
-import com.tft_mvvm.app.ui.adapter.MyAdapter
-import com.tft_mvvm.app.ui.OnItemClickListener
+import androidx.viewpager.widget.ViewPager
+import com.tft_mvvm.app.ui.adapter.ViewPagerAdapter
+import com.tft_mvvm.app.ui.fragment.ShowChampByGold
+import com.tft_mvvm.app.ui.fragment.ShowChampByName
+import com.tft_mvvm.app.ui.fragment.ShowRecommendTeam
 import com.tft_mvvm.champ.R
 import kotlinx.android.synthetic.main.activity_main.*
-import org.koin.androidx.viewmodel.ext.android.viewModel
 
-class MainActivity : AppCompatActivity(),
-    OnItemClickListener {
-    private val champViewModel: MainViewModel by viewModel()
-    private lateinit var adapter: MyAdapter
-    private lateinit var champs: ArrayList<Champ>
-    private var isLoading: Boolean? = null
+class MainActivity : AppCompatActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
-        setupUI()
-        getChamps()
-        sort()
-
-    }
-
-    private fun setupUI() {
-        adapter = MyAdapter(arrayListOf(),this)
-        rv.layoutManager = GridLayoutManager(this, 4)
-        rv.adapter = adapter
-        swipe_refresh_layout.setOnRefreshListener {
-            getChamps()
-            group_radio_btn.clearCheck()
-            swipe_refresh_layout.isRefreshing = isLoading!!
-        }
-    }
-
-    private fun sort() {
-        btn_sort_by_name.setOnClickListener {
-            if(isLoading==false){
-                adapter.sortByName()
-            }
-        }
-        btn_sort_by_coat.setOnClickListener {
-            if(isLoading==false){
-                adapter.sortByCoat()
-            }
-        }
-    }
-
-    private fun getChamps() {
-        champs = ArrayList(arrayListOf())
-        champViewModel.getChamps()
-        champViewModel.getChampsLiveData()
-            .observe(this, Observer {
-                champs.clear()
-                champs.addAll(it)
-                adapter.addData(champs)
-            })
-        champViewModel.isRefresh()
-            .observe(this, Observer { isLoading = it })
+        addTab(view_pager)
+        tab_layout.setupWithViewPager(view_pager)
     }
 
-    override fun onClickListener(champ: Champ) {
-        val intent : Intent = Intent(this,
-            DetailsChamp::class.java)
-            intent.putExtra("champ",champ)
-        startActivity(intent)
+    private fun addTab(viewPager: ViewPager){
+        val showChampByGold = ShowChampByGold()
+        val showChampByName = ShowChampByName()
+        val showRecommendTeam = ShowRecommendTeam()
+        val adapter = ViewPagerAdapter(supportFragmentManager, arrayListOf(), arrayListOf())
+        adapter.add(fragment = showChampByGold, title = "Gold")
+        adapter.add(fragment = showChampByName, title = "Gold")
+        adapter.add(fragment = showRecommendTeam, title = "Gold")
```

Comment:
`title = "Gold"` 
You should avoid hardcode here. Using Enum or Object class to resolve it.